Add typecast to native border router switch statement

tonnenpinguin opened the pull request:
When running the native border router on my raspbian installation it would always fail because for some reason the while loop would not consider the statement to be true.
My best guess would be that the -1 value for a char is `0xFF` while the same for an int would be `0xFFFFFFFF`.

nfi commented:
Good catch but I think the problem is caused by `c` being a `char` instead of an `int`. `getopt()` returns an int so I think a nicer solution would be to change `c` to be an int instead of a char.

g-oikonomou commented:
Also see #982

tonnenpinguin commented:
Fair enough. I changed the type of c to int and removed the typecast. I guess that works around the char signedness discussion and as @nfi mentioned should work for linux targets.

joakimeriksson commented:
This is a fix that we have used since long at Yanzi and in other Contiki projects. So this is a valid fix 👍 (also very easy to review ;-)

alignan commented:
If you squash your commits then :+1:

tonnenpinguin commented:
Done!

nvt commented:
:+1:
